pkg/controller/perconaservermongodb/version.go

```diff
@@ -189,9 +189,27 @@ func majorUpgradeRequested(cr *api.PerconaServerMongoDB, fcv string) (UpgradeReq
 		return UpgradeRequest{true, apply, ver}, nil
 	}
 
-	can, err := canUpgradeVersion(fcv, ver)
+	mongoVer, err := v.NewSemver(cr.Status.MongoVersion)
 	if err != nil {
-		return UpgradeRequest{false, "", ""}, errors.Wrap(err, "can't upgrade")
+		return UpgradeRequest{false, "", ""}, errors.Wrap(err, "faied to make semver")
+	}
+
+	newMM := MajorMinor(newVer)
+	mongoMM := MajorMinor(mongoVer)
+
+	can := false
+
+	if newMM > mongoMM {
```

Comment:
now we have comparison logic in two places :( one in `canUpgradeVersion` and another one in majorUpgradeRequested 😞 
can we handle logic in one place?

Reply:
But the comparisons are different.
Here we check if user want to update or downgrade.
There we check if we can upgrade from one version to another.